### PR TITLE
Remove unknown rules from managed security groups on openstack

### LINKED ISSUE #9716: Have kops ensure the correct rules for the master/node security groups
**1. Describe IN DETAIL the feature/behavior/change you would like to see.**

Right now, kops does not clean up old security group rules or rules added manually. There is no reason why kops shouldn't do this since we support adding additional security groups to be used for whatever other purposes.

**2. Feel free to provide a design supporting your feature request.**

The larger, but perhaps better way of solving this is to make security group rules a property of the securitygroup task. Then `Find()` can just get the current set of rules and ensure they match expected. There are some rule deletions that take place there already, for example if one enable bastions. But there is very specific logic around those few ports.

The more direct way of doing this is implement deletions in the securitygrouprule task. E.g add a Delete property to the task, create a task for every existing rule in the security group. If not an identical rule is created in `firewall.go`, set the Deletion property to true. Then one could also remove the `FindDeletions` logic for rules in the securitygroup task. 

Any thoughts?

### PR DESCRIPTION
**Pull request by olemarkus**
Fixes #9716 for openstack.

### CODE REVIEW

File: pkg/model/openstackmodel/firewall.go [hunk 1]
```diff
@@ -465,15 +474,90 @@ func (b *FirewallModelBuilder) addProtokubeRules(c *fi.ModelBuilderContext, sgMa
...
+	for sgName, sgt := range sgMap {
+
+		sgs, err := osCloud.ListSecurityGroups(sg.ListOpts{
+			Name: sgName,
+		})
+		if err != nil {
+			return err
+		}
+		if len(sgs) != 1 {
+			continue
```
Comment (zetaab) [inline pkg/model/openstackmodel/firewall.go:501]: if we have two sgs with same name? in that case imo we should print log about that

**Review by zetaab — CHANGES_REQUESTED** [review]
did you test this against running openstack kops cluster? if yes, then it looks quite good

File: pkg/model/openstackmodel/firewall.go (hunk 1 above)
Comment (olemarkus) [reply pkg/model/openstackmodel/firewall.go:501]: Erroring out now if that is the case. Thanks for catching.

**Comment by olemarkus** [comment]
Yeah, tested that extra rules are detected and removed. And that at least on my test setups, there are no unwanted removals.

**Review by zetaab — APPROVED** [review]
thanks!

/approve
/lgtm

**Pull request merged**